spec.md

```diff
@@ -5,76 +5,63 @@ keywords: registry, on-prem, images, tags, repository, distribution, api, advanc
+	- [Historical Context](#historical-context)
+- [Definitions](#defintions)
+	- [Common Terms](#introduction)
+	- [Document Language](#document-language)
+- [Conformance](#conformance)
+	- [Minimum Requirements](#minimum-requirements)
+	- [Official Certification](#official-certification)
+	- [Workflow Categories](#workflow-categories)
+		1. [Pull](#pull)
+		2. [Push](#push)
+		3. [Content Discovery](#content-discovery)
+		4. [Content Management](#content-management)
+- [HTTP API](#http-api)
+	- [Endpoints](#endpoints)
+	- [Error Codes](#error-codes)
 
-The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119) (Bradner, S., "Key words for use in RFCs to Indicate Requirement Levels", BCP 14, RFC 2119, March 1997).
+## Overview
 
-The key words "unspecified", "undefined", and "implementation-defined" are to be interpreted as described in the [rationale for the C99 standard][c99-unspecified].
+### Introduction
 
-An implementation is not compliant if it fails to satisfy one or more of the MUST, MUST NOT, REQUIRED, SHALL, or SHALL NOT requirements for the protocols it implements.
-An implementation is compliant if it satisfies all the MUST, MUST NOT, REQUIRED, SHALL, and SHALL NOT requirements for the protocols it implements.
+The Open Container Initiative Distribution Specification defines an API protocol to facilitate and standardize the distribution of content, especially related to container images and associated artifacts.
+
+### Historical Context
 
-## Historical Context
+The spec is based on the specification for the [Docker Registry HTTP API V2 protocol](https://github.com/docker/distribution/blob/5cb406d511b7b9163bff9b6439072e4892e5ae3b/docs/spec/api.md).
 
 For relevant details and a history leading up to this specification, please see the following issues:
 
 - [moby/moby#8093](https://github.com/moby/moby/issues/8093)
 - [moby/moby#9015](https://github.com/moby/moby/issues/9015)
 - [docker/docker-registry#612](https://github.com/docker/docker-registry/issues/612)
 
-<!--- TODO: add relevant background information here --->
+## Definitions
+
+### Common Terms
+
+Several terms are used frequently in this document and warrant basic definitions:
+
+- **Registry**: a HTTP service which implements this spec
+- **Client**: a tool that communicates with registries over HTTP
+- **Push**: the act of uploading content to a registry
+- **Pull**: the act of downloading content from a registry
+- **Artifact**: a single piece of content, made up of a manifest and one or more layers
+- **Manifest**: a JSON document which defines an artifact
+- **Layer**: a single part of all the parts which comprise an artifact
+- **Config**: a special layer defined at the top of a manifest containing artifact metadata
+- **Blob**: a single binary content stored in a registry
```

Comment:
Do we want to distinguish between "Layer" and "Blob" here?

Reply:
@rchincha how would you best describe the difference?

Reply:
the definitions appear to be circular, use words not defined.. For example:

layer is a part in parts (part not defined) which comprises an artifact..

artifact is a piece of content (content is content that is addressable and can be downloaded choose one: manifest or blob) made up of a manifest and one or more layers (one or more undefined parts)

blob is binary content (content is content -- either manifest or blob)

Reply:
A blob is the binary form of metadata OR [most commonly] content that is stored by the registry.
An artifact is one conceptual piece of content stored as blobs with an accompanying manifest containing metadata.
A layer is one kind of blob used for container image artifacts

Reply:
@jzelinskie vaguely remember you had a carefully compiled set of definitions. Let's look at that and see if we can just reuse that prose.